services/graph/pkg/service/v0/sharedbyme.go

```diff
@@ -0,0 +1,232 @@
+	}
+	driveItems, err = g.cs3UserSharesToDriveItems(ctx, lsUserSharesResponse.Shares, driveItems)
+	if err != nil {
+		return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+	}
+	return driveItems, nil
+}
+
+func (g Graph) listPublicShares(ctx context.Context, driveItems driveItemsByResourceID) (driveItemsByResourceID, error) {
+
+	gatewayClient, err := g.gatewaySelector.Next()
+	if err != nil {
+		g.logger.Error().Err(err).Msg("could not select next gateway client")
+		return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+	}
+
+	filters := []*link.ListPublicSharesRequest_Filter{}
+
+	req := link.ListPublicSharesRequest{
+		Filters: filters,
+	}
+
+	lsPublicSharesResponse, err := gatewayClient.ListPublicShares(ctx, &req)
+	if err != nil {
+		return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+	}
+	if lsPublicSharesResponse.Status.Code != rpc.Code_CODE_OK {
+		return driveItems, errorcode.New(errorcode.GeneralException, lsPublicSharesResponse.Status.Message)
+	}
+	driveItems, err = g.cs3PublicSharesToDriveItems(ctx, lsPublicSharesResponse.Share, driveItems)
+	if err != nil {
+		return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+	}
+	return driveItems, nil
+
+}
+
+func (g Graph) cs3UserSharesToDriveItems(ctx context.Context, shares []*collaboration.Share, driveItems driveItemsByResourceID) (driveItemsByResourceID, error) {
+	for _, s := range shares {
+		g.logger.Debug().Interface("CS3 UserShare", s).Msg("Got Share")
+		resIDStr := storagespace.FormatResourceID(*s.ResourceId)
+		item, ok := driveItems[resIDStr]
+		if !ok {
+			itemptr, err := g.getDriveItem(ctx, storageprovider.Reference{ResourceId: s.ResourceId})
+			if err != nil {
+				g.logger.Debug().Err(err).Interface("Share", s.ResourceId).Msg("could not stat share, skipping")
+				continue
+			}
+			item = *itemptr
+		}
+		perm := libregraph.Permission{}
+		perm.SetRoles([]string{"TODO:insert-role-here"})
+		perm.SetId(s.Id.OpaqueId)
+		grantedTo := libregraph.SharePointIdentitySet{}
+		switch s.Grantee.Type {
+		case storageprovider.GranteeType_GRANTEE_TYPE_USER:
+			gatewayClient, err := g.gatewaySelector.Next()
+			if err != nil {
+				g.logger.Error().Err(err).Msg("could not select next gateway client")
+				return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+			}
+			req := user.GetUserRequest{
+				UserId: s.Grantee.GetUserId(),
+			}
+			res, err := gatewayClient.GetUser(ctx, &req)
+			if err != nil {
+				return driveItems, errorcode.New(errorcode.GeneralException, err.Error())
+			}
+			user := libregraph.NewIdentityWithDefaults()
+			user.SetId(s.Grantee.GetUserId().GetOpaqueId())
+			switch res.Status.Code {
+			case rpc.Code_CODE_OK:
+				cs3User := res.GetUser()
+				user.SetDisplayName(cs3User.GetDisplayName())
+			case rpc.Code_CODE_NOT_FOUND:
+				g.logger.Warn().Str("userid", s.Grantee.GetUserId().GetOpaqueId()).Msg("User not found by id")
```

Comment:
Return an error? Avoid calling `grantedTo.SetUser(*user)`?

Reply:
:thinking: returning an error is probably not right. That would result in the GetSharedByMe to error out just because a single grantee does not exist anymore. 

IMO we should just log a warning (like a do now). Though it is debatable if we should return a permission with just the userid (and no displayname) or if we should skip grantees in this case.